widget/tabcontainer.go

```diff
@@ -95,6 +98,11 @@ func (t *TabContainer) SelectTabIndex(index int) {
 	r := cache.Renderer(t).(*tabContainerRenderer)
 	r.Layout(t.size)
 	t.Refresh()
+
+	callback := t.OnTabSelected
```

Comment:
This could be inlined:
```
if callback := t.OnTabSelected; callback != nil {
  callback(t.Items[t.current])
}
```

Reply:
I’m not really a fan of setting up inside an if like this - unless it’s an ok check for casting. Either way in this instance you don’t need the variable set anyhow :)

Reply:
Fair enough, this is shorter anyway;
```
if t.OnTabSelected != nil {
  t.OnTabSelected(t.Items[t.current])
}
```

Reply:
I thought about this too. But I have a question: is it possible a race condition between checking for nil and calling a function? I understand that we still have no lock on set/read, but still... I think I've had this problem once before.

Reply:
If there is a race condition on read then setting it to another variable sadly does not remove that. I appreciate that it’s “less likely” but technically still there unless you guard access to the variable.

Reply:
Sorry, I was not clear, I meant to protect only one special case when at one time function was set and later on it set as nil and calling nil func after `if` will cause panic.

Just out of curiosity I tried to reproduce this case and failed, probably on a low level it doesn't work like that.